libff/algebra/fields/tests/test_fields.cpp

```diff
@@ -258,6 +259,124 @@ void test_Fp12_2over3over2_mul_by_024()
     ASSERT_EQ(result_slow, result_mul_024);
 }
 
+void
+test_field_get_digit_alt_bn128()
+{
+    using FieldT = Fr<alt_bn128_pp>;
+
+    {
+        // 0x30644e72e131a029b85045b68181585d2833e84879b9709143e1f593f0000000
+        const auto value = FieldT(-1).as_bigint();
+
+        // 127 -> 0, 126 -> 3, 125 -> 0
+        ASSERT_EQ(0, field_get_digit(value, 2, 127));
+        ASSERT_EQ(3, field_get_digit(value, 2, 126));
+        ASSERT_EQ(0, field_get_digit(value, 2, 125));
+    }
+
+    {
+        // 0x30644e72e131a029b85045b68181585d2833e84879b9709143e1f593f0000000
+        const auto value = FieldT(-1).as_bigint();
+
+        ASSERT_EQ(0x3064, field_get_digit(value, 16, 15));
+        ASSERT_EQ(0x4e72, field_get_digit(value, 16, 14));
+        ASSERT_EQ(0xe131, field_get_digit(value, 16, 13));
+        ASSERT_EQ(0xa029, field_get_digit(value, 16, 12));
+        ASSERT_EQ(0xb850, field_get_digit(value, 16, 11));
+        ASSERT_EQ(0x45b6, field_get_digit(value, 16, 10));
+        ASSERT_EQ(0x8181, field_get_digit(value, 16, 9));
+        ASSERT_EQ(0x585d, field_get_digit(value, 16, 8));
+        ASSERT_EQ(0x2833, field_get_digit(value, 16, 7));
+        ASSERT_EQ(0xe848, field_get_digit(value, 16, 6));
+        ASSERT_EQ(0x79b9, field_get_digit(value, 16, 5));
+        ASSERT_EQ(0x7091, field_get_digit(value, 16, 4));
+        ASSERT_EQ(0x43e1, field_get_digit(value, 16, 3));
+        ASSERT_EQ(0xf593, field_get_digit(value, 16, 2));
+        ASSERT_EQ(0xf000, field_get_digit(value, 16, 1));
+        ASSERT_EQ(0x0000, field_get_digit(value, 16, 0));
+    }
+
+    {
+        // 0x30644e72e131a029b85045b68181585d2833e84879b9709143e1f593f0000000
+        const auto value = FieldT(-1).as_bigint();
+
+        ASSERT_EQ(0x000, field_get_digit(value, 12, 0));
```

Comment:
This ASSERT is duplicated (see at the bottom of your block). Since you navigate your digit from left to right, this line must be deleted (the final line of the block kept)

Reply:
Looks like that was left from debugging.  Should be removed now.